### PR TITLE
Doc and name improvements

### PR DESCRIPTION
**Pull request by unscriptable**
I hope this helps people understand and *use* this library!  I think it could be extremely useful for TetraScience.

- [x] More examples
- [x] General philosophy
- [x] General and detailed usage
- [x] Better names for things like `"$spread"`, `"$if"`, and `"$map"` and `transform` (which might be better called `evaluate`?).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,48 +52,185 @@ getInputFile()
...
 5.  Conditional output can be achieved for many cases with the `"$if"` key.
     The value of the `"$if"` key is a JsonPath query.  If the query finds
     any matching value(s) in the input document, the entire template object is
     used.  Otherwise, it is omitted.
 
+### Possible future features
+
+1.  JsonPath allows the input document arrays to be filtered via simple
+    expressions.  This could be valuable.
+1.  JsonPath allows "scoped" queries via a leading `"@"` instead of a leading
```
Comment (joegoldbeck) [inline README.md]: Maybe this should be `2.`?

Comment (unscriptable) [reply README.md]: Markdown will auto-number when rendered, but , yah, it looks like a mistake when looking at the source.

**Comment by unscriptable** [comment]
I want to rename a few things to make them less confusing.  This will cause breaking changes to the API.  Just wondering if we should retag 1.0.0 rather than create 2.0.0?

**Comment by joegoldbeck** [comment]
Go ahead with the renaming! Let's make it `2.0.0` though, since `1.0.0` is already referenced :)

**Comment by unscriptable** [comment]
I kept `$map` for now.  Not sure it'll make sense to any non-FP folks, though.  Any ideas?

I changed `$spread` to `$each` and `$if` to `$exists` (which is semantically what it does).

**Comment by joegoldbeck** [comment]
👍 

I think $map is fine, though I hear what you're saying :)

**Pull request merged**